document/workflow/variable-tree/data-source/skeleton-tree-data-source.ts

```diff
@@ -222,14 +241,11 @@ export class SkeletonTreeNode extends Comparable implements Traversable, Graphic
     /**
      * @inheritdoc
      */
-    match(path: ComparablePath): Traversable {
-        if (this.equals(path)) {
-            let matchingNode: Traversable;
+    match(path: ComparablePath): SkeletonTreeNode {
+        if (this.getXFLExpression() === path.path) {
+            let matchingNode: SkeletonTreeNode;
             if (path.child) {
-                let i = 0;
-                while (i < this.children.length && !(matchingNode = this.children[i].match(path.child))) {
-                    i++;
-                }
+                this.children.find(node => matchingNode = node.match(path.child));
```

Comment:
`node => !!(matchingNode = node.match(path.child))`

might be cleaner and easier to understand.